File: app/src/main/java/com/kaciula/archiman/presentation/screen/userdetails/UserDetailsPresenter.java
package com.kaciula.archiman.presentation.screen.userdetails;

import com.kaciula.archiman.domain.util.SchedulerProvider;
import com.kaciula.archiman.presentation.screen.home.UserViewModel;

class UserDetailsPresenter implements UserDetailsContract.Presenter {

  private final UserDetailsContract.View view;
  private final UserViewModel user;
  private final SchedulerProvider schedulerProvider;

  UserDetailsPresenter(UserDetailsContract.View view, UserViewModel user,
      SchedulerProvider schedulerProvider) {
    this.view = view;
    this.user = user;
    this.schedulerProvider = schedulerProvider;
  }

  @Override
  public void init() {
    view.render(UserDetailsViewModel.create(user.name()));
  }

  @Override
  public void start() {

  }

  @Override
  public void stop() {

  }

  @Override
  public void destroy() {

  }
}

File: app/src/main/java/com/kaciula/archiman/presentation/screen/userdetails/UserDetailsPresenter.kt
package com.kaciula.archiman.presentation.screen.userdetails

import com.kaciula.archiman.domain.util.SchedulerProvider
import com.kaciula.archiman.presentation.screen.home.UserViewModel

class UserDetailsPresenter(private val view: UserDetailsContract.View,
                           private val user: UserViewModel,
                           private val schedulerProvider: SchedulerProvider) : UserDetailsContract.Presenter {

    override fun init() {
        view.render(UserDetailsViewModel(user.name))
    }

    override fun start() {}

    override fun stop() {}

    override fun destroy() {}
}
